test/integration/targets/openssl_certificate/tests/validate.yml

```diff
@@ -0,0 +1,23 @@
+- name: Validate certificate (test - privatekey modulus)
+  shell: 'openssl rsa -noout -modulus -in {{ output_dir }}/privatekey.pem | openssl md5'
+  register: privatekey_modulus
+
+- name: Validate certificate (test - certificate modulus)
+  shell: 'openssl x509 -noout -modulus -in {{ output_dir }}/cert.pem | openssl md5'
+  register: cert_modulus
+
+- name: Validate certificate (test - subject)
+  shell: "openssl x509 -noout -subject -in {{ output_dir }}/cert.pem -nameopt oneline,-space_eq"
+  register: cert_subject
+
+- name: Validate certificate (test - issuer)
+  shell: "openssl x509 -noout -issuer -in {{ output_dir }}/cert.pem -nameopt oneline,-space_eq"
+  register: cert_issuer
+
+
+- name: Validate certificate (assert)
+  assert:
+    that:
+      # Certificate is selfsigned: subject == issuer)
```

Comment:
IIUC [RFC3280](https://tools.ietf.org/html/rfc3280) (section6.1)

> A certificate is self-issued if the DNs that appear in the subject
>   and issuer fields are identical and are not empty

So what I understand is that this is true for 'self-issued' certificate, but not a requirement for 'self-signed' certificate.

Hence in this scenario, the ouput of `openssl x509 -noout -issuer -in {{ output_dir }}/cert.pem -nameopt oneline,-space_eq` is `issuer=` which I think is valid.

However, we could add an option to the self-signed backend to eventually allow one to specify the issuer via [set_issuer()](https://pyopenssl.org/en/stable/api/crypto.html#OpenSSL.crypto.X509.set_issuer)

What are your thoughts on this ?

Reply:
That's definitely an option that is fine by me, it seemed just like an easy assertion to initilly try out. I'm more concerned about the CI failure when checking the private key in the module itself.

Reply:
I don't unserstand: 

> I'm more concerned about the CI failure when checking the private key in the module itself

The following assertion works:

```
 - name: Validate certificate (assert)      
    assert:                                  
        that:                                                                          
          - cert_modulus.stdout == privatekey_modulus.stdout 
```

What are you exactly refering to ?

Reply:
https://app.shippable.com/github/ansible/ansible/runs/36244/21/console for example

Reply:
@MarkusTeufelberger ok, I found the root cause issue. I'll provide a patch as soon as I find an elegant solution for it.

Reply:
@MarkusTeufelberger could you integrate this commit https://github.com/Spredzy/ansible/commit/a25db3d61f69a6d5f82ce67d7ed2abff199bb539 in your PR (if you agree with its content - else let me know if there is anything you'd like to see modified)?

This will fix the issue you've spotted. Once this PR merged, we will need to backport it for RC2 since you've found an actual issue with the code. Thanks again for that.

Reply:
Are dictionary parameters (subject, issuer) internally parsed into tuples? Otherwise it might be necessary to add a `elif isinstance(attr, dict):` line there, right?

Reply:
Yep, they are internally parsed into tuple. Hence the tuple test. None of the current input are parsed as dict. so that why the test isn't present.

Reply:
Great, thanks! :-)

Reply:
Ideally, you should put commit a4026d0 before c66ccce in your branch so tests are always ok no matter on which commit it is run. Also for now I think we can drop the subject.stdout == issuer.stdout until we add this feature in the code.